src/numerics.jl

```diff
@@ -0,0 +1,326 @@
+using UnPack
+using SparseArrays
+using LinearAlgebra
+
+include("Klimaparameter.jl")
+
+#Mapping functions
+##################
+function index2d(k,nx)
+    j = floor(Int,(k-1)/nx)+1
+    # TODO: add consistency check with ny?
+    return k-(j-1)*nx,j 
+end
+
+function index1d(i,j,nx)
+    # TODO: add consistency check with ny?
+    return i+(j-1)*nx
+end
+
+struct mesh_t
+    nx::Int         # Number of DOFs in x (Longitude)
+    ny::Int         # Number of DOFs in y (Latitude)
+    dof::Int
+    h::Float64
+    sh2::Float64
+    geom::Float64
+    csc2::Array{Float64,1}
+    cot::Array{Float64,1}
+    area::Array{Float64,1}
+end
+
+function mesh_init(nx::Int)
+    ny = Int(nx/2+1) # 65
+    dof = nx*ny
+
+    h = pi / (ny-1) # Uniform grid size in radians
+    sh2 = 1 / h^2
+
+    # Trigonometric definitions for inner degrees of freedom
+    csc2 = zeros(Float64,ny) # csc^2(theta), where theta is the colatitude angle (from the pole)
```

Comment:
`zeros` and virtually all other array-creating functions default to `Float64`, so I recommend to omit this everywhere.
```suggestion
    csc2 = zeros(ny) # csc^2(theta), where theta is the colatitude angle (from the pole)
```